dev/com.ibm.ws.kernel.metatype.helper/src/com/ibm/wsspi/config/internal/FilesetImpl.java

```diff
@@ -541,4 +573,29 @@ private void notifyListeners() {
         }
     }
 
+    @Override
+    @FFDCIgnore(Exception.class)
+    public String toString() {
+        try {
+
+            StringBuilder sb = new StringBuilder();
+
+            sb.append("FileSet for directory: " + basedir);
+            sb.append(System.lineSeparator());
+
+            if (filter.hasFilters()) {
+                sb.append(filter.getHumanReadableSummary()); //provides its own line break
+            }
+
+            String files = fileset.stream().map(File::getPath).collect(Collectors.joining(", "));
+            sb.append("Which found the following files: " + files);
+
+            return sb.toString();
+
+        } catch (Exception e) {
+            String originalToString = super.toString();
+            return "Caught an exception calling to String on " + originalToString + " exception was: " + e.toString();
```

Comment:
issue: loses all progress because of one exception
suggestion: add exception report (maybe with a newline) to the string builder instead